### DISCUSSION TITLE
What UX problems have you encountered with Lightning?

### DISCUSSION THREAD

**Discussion by GBKS** [BitcoinDesign/Guide#432] [General]
Expiring invoices were mentioned as a problem a few times during [Community Call #18](https://github.com/BitcoinDesign/Meta/issues/97). What else have you come across?

Here's an error @ConorOkus encountered when trying to [buy a t-shirt](https://scarce.city/sales/ZETRA.T111).

Paid using an on-chain transaction in the end when the intention was to use lightning.

![Screenshot_20210709-172949_Muun 1](https://user-images.githubusercontent.com/695901/125774794-e426cdea-390b-4348-aa35-d1dca8b97813.png)

**Comment by psacramento-gh** [comment]
![Screenshot_20210715-135627.png](https://user-images.githubusercontent.com/76049494/125784348-ef95f0b0-b7fc-479c-8c70-523584f6f0ec.png)

**Reply by GBKS** [reply]
Interesting... do you have the link for that post at the bottom?

**Comment by Bosch-0** [comment]
Dynamic commit fees can be a surprising event for people managing their own channels: https://www.reddit.com/r/Bitcoin/comments/cjtgf6/lightning_network_question_commit_fee/

The commit fee is some satoshi's your channel reserves so it can be closed when the time comes to close it. 

![image](https://user-images.githubusercontent.com/55287964/125881608-6c8663d3-3afa-4fa9-881a-1e970c794130.png)

**Comment by shawnyeager** [comment]
Not knowing if it’s beneficial or possible to activate multi-path payments. I went to pay an invoice yesterday (~150k sats) with Zeus wallet. I was shown a prediction of ~18% likelihood of success, so I attempted MPP. Failed twice. I then attempted a standard payment, which completed almost immediately.

**Reply by pavlenex** [reply]
### Separate networks

Just onboarded someone who's never heard of Bitcoin yesterday via Muun, the fact Bitcoin and Lightning were separate networks that was very hard for the person I onboarded to understand. I sent them funds via LN then they wanted to send back via BTC which was a separate tab.

**Reply by pavlenex** [reply]
Incoming channels and liquidity for users running their own nodes is always a challenge. They are usually referred to bitrefill, lnbig etc in order to get started. In BTCPay we're trying to find a ways on how to offer these different providers via UI, but also communicate it to the user easily enough on why they may or may not want to use a provider for an immediate incoming liquidity.

**Reply by pavlenex** [reply]
Sorry, didn't mean to reply this in a thread GitHub for mobile is weird.

**Reply by moneyball** [reply]
I wonder why Zeus asks if MPP should be used

**Reply by shawnyeager** [reply]
I'm unsure. It seems to be an option for all sends. Here's a screenshot of an invoice payment I just set up in Zeus. Also—and with full props to Evan, because I really like Zeus—the probability seems way off. The below invoice is from custodial (default) Blue Wallet, so I would assume it's almost guaranteed to succeed.

![IMG_4661](https://user-images.githubusercontent.com/980297/125972957-ba46f33b-55d8-4dd7-aa95-b8ee77caffbb.JPG)

**Reply by GBKS** [reply]
I forwarded your comment to Evan, he said he'll look into it.

**Comment by maltokyo** [comment]
Few from me:
- The entire concept of channels, why they are needed, I would say most people give up at this stage.
- What are commit fees (and lack of any semblance of control over how much they are going to be.  Most people open a small channel at the start once they work out the channel thing, and then are astounded at the %age of the channel amount locked up in fees.  They don't understand why this happens, and then they never revisit lightning again as it is "too expensive to open channels".)
- Failed payments (couldn't find a route etc..) leave users with "wtf!?" as their user experience.  Suggestions on what is needed to make the payment work are what they need, not error messages with no clear next step.
- If the user runs own node, how to ensure backup of the channel state is not clear for most node setups at the moment, putting funds at risk.

**Comment by johnsBeharry** [comment]
On the lightning-dev list i just came across someone trying to propose an on-chain fallback address for lightning invoices. This already exists in the BOLT11 spec though but the problem with routing they encountered is one that is worth exploring design solutions for. I've documented some of this [here](https://github.com/peakshift/bitcoin-ux/blob/master/payments/qr-codes.md#bolt11-invoices-encoded-in-bip21-uris) even though the emphasis was on using BIP21 URIs (because of support for non-lightning applications, idealy everything would default to BOLT11, even on-chain as it's a more expressive format). 

![image](https://user-images.githubusercontent.com/183140/125989454-88904df6-dbc8-4d52-ad79-6ef7b7bda75d.png)

In the graphic above the sender scans a QR code that contains a lightning invoice, and a bitcoin on-chain address. The wallet by default attempts to make a lightning payment but encounters a routing error. The sender can at this point choose to make the payment on-chiain instead (or to open a channel with the receiver?).

**Reply by shawnyeager** [reply]
Is this the same technique that Muun is using?

**Reply by Bosch-0** [reply]
An interesting UX possibility that this concept could open up is the ability to have one QR code rather than separate receive QRs for LN invoices / on-chain addresses (A BLIP standard maybe?). For more technical focused wallets having them separate would probably be preferred but for a average user wallet like Muun it would be more user friendly imo.

I know when I've on boarded people into Bitcoin through Muun the first usability issue they experience is why is there a on-chain and Lightning method of receiving a payment? This could somewhat resolve that issue. You could have it so when this 'One QR' is shared to the sender and they scan it they get the option to either send via the on-chain address or the LN invoice (depending on what wallet they are using / what layer they hand funds on etc.) 

Users who have already received Bitcoin are likely more familiar with the two concepts thus this extra burden falling on them (having to pick which option to pay with) rather than a user who has never received a payment would lead to less on boarding friction which is critical for user retention. 

You could even take this a step further and on the sending side have the two options as something like 'Send payment' for on-chain and 'Send instant payment' for the LN option.

**Comment by GBKS** [comment]
A few comments from Twitter to better keep track of them:
- Errors messaging is too obscure and honestly makes no sense to general users. Errors need to be more informative of what is actually happening and as "non-technical" as possible. ([link](https://twitter.com/olthoff/status/1417536543124721665?s=20))
- Usable/Actionable information when payment fails. Help me understand what to change for the next attempt. ([link](https://twitter.com/statusquont/status/1416154634339733507?s=20))
- Getting channel liquidity both ways is still a pain in the ass. When a channel/LN app malfunctions I have no idea why & it takes a lot of effort to figure it out. (I’ve tried 8 ish & never had 1 work smoothly on the first try.) Stability requires a VPS, otherwise constant fixing. ([link](https://twitter.com/La__Cuen/status/1417525048978509825?s=20))

**Reply by shawnyeager** [reply]
I think micro-copy/UX copy is a big area of opportunity. It’s a domain of its own, but maybe something like [this](https://luna.sainsburys.co.uk/copy/writing-principles) has a place in a design guideline.

**Comment by pavlenex** [comment]
> I had 40K sats in a MuunWallet I wanted to send to another wallet. It was cheaper to send on chain than via LN. Muun calculated my LN fees at nearly 700 sats! On chain was cheaper and went through in next block. Sorry routing node operators, miners got that fee.

https://twitter.com/EndlessBullrun/status/1423246882202030081?s=20

Possibly related to submarine swaps Muun is using for easier onboarding.

https://twitter.com/EndlessBullrun/status/1423246882202030081?s=20

**Reply by Bosch-0** [reply]
On-chain has been so cheap lately I'd imagine many people have been experiencing this issue.
